tool/icons/update_icons.dart

```diff
@@ -1,105 +1,122 @@
+      'CupertinoIcons', 'cupertino');
+
+  // tell the user how to generate the icons
+  print('');
+  print("In order to re-generate the icons, open the iOS Simulator, and "
+      "'flutter run' from the tool/icon_generator/ directory.");
+}
+
+Future<String> downloadUrl(String url) async {
+  HttpClientRequest request = await new HttpClient().getUrl(Uri.parse(url));
+  HttpClientResponse response = await request.close();
+  List<String> data = await response.transform(utf8.decoder).toList();
+  return data.join('');
+}
 
-  final String iconData = iconFile.readAsStringSync();
-  final String codepointData = codepointsFile.readAsStringSync();
-  final String newIconData = regenerateIconsFile(iconData, codepointData);
+// static const IconData threesixty = IconData(0xe577,
+final RegExp regexp =
+    new RegExp(r'static const IconData (\S+) = IconData\(0x(\S+),');
 
-  if (argResults[kOptionDryRun])
-    stdout.writeln(newIconData);
-  else
-    iconFile.writeAsStringSync(newIconData);
+List<Icon> parseIconData(String data) {
+  return regexp.allMatches(data).map((Match match) {
+    return Icon(match.group(1), int.parse(match.group(2), radix: 16));
+  }).toList();
 }
 
-String regenerateIconsFile(String iconData, String codepointData) {
-  final StringBuffer buf = new StringBuffer();
-  bool generating = false;
-  for (String line in LineSplitter.split(iconData)) {
-    if (!generating) buf.writeln(line);
-    if (line.contains(kBeginGeneratedMark)) {
-      generating = true;
-      final String iconDeclarations = generateIconDeclarations(codepointData);
-      buf.write(iconDeclarations);
-    } else if (line.contains(kEndGeneratedMark)) {
-      generating = false;
-      buf.writeln(line);
+void generateProperties(List<Icon> icons, String filename, String pathSegment) {
+  StringBuffer buf = StringBuffer();
+  buf.writeln('# Generated file - do not edit.');
+  buf.writeln();
+  buf.writeln('# suppress inspection "UnusedProperty" for whole file');
+
+  Set<int> set = new Set();
+
+  for (Icon icon in icons) {
+    buf.writeln();
+
+    if (set.contains(icon.codepoint)) {
+      buf.write('# ');
     }
+
+    buf.writeln('${icon.codepoint.toRadixString(16)}.codepoint=${icon.name}');
+    buf.writeln('${icon.name}=/flutter/$pathSegment/${icon.name}.png');
+
+    set.add(icon.codepoint);
   }
-  return buf.toString();
+
+  new File(filename).writeAsStringSync(buf.toString());
+
+  print('wrote $filename');
 }
 
-String generateIconDeclarations(String codepointData) {
-  return LineSplitter
-      .split(codepointData)
-      .map((String l) => l.trim())
-      .where((String l) => l.isNotEmpty)
-      .map(getIconDeclaration)
-      .join();
+void generateDart(
+    List<Icon> icons, String filename, String prefix, String import) {
+//  final List<IconData> icons = [
+//    Icons.visibility
+//  ];
+
+  StringBuffer buf = StringBuffer();
+  buf.writeln('// Generated file - do not edit.');
```

Comment:
nit: using ''' here would make the generated code a little easier to edit if needed.

Reply:
done